examples/java/src/test/java/org/apache/beam/examples/snippets/SnippetsTest.java

```diff
@@ -154,6 +156,73 @@ public void testCoGroupByKeyTuple() throws IOException {
     p.run();
   }
 
+  /* Tests SchemaJoinPattern */
+  @Test
+  public void testSchemaJoinPattern() {
+    // [START SchemaJoinPatternCreate]
+    // Define Schemas
+    Schema emailSchema =
+        Schema.of(
+            Schema.Field.of("name", Schema.FieldType.STRING),
+            Schema.Field.of("email", Schema.FieldType.STRING));
+
+    Schema phoneSchema =
+        Schema.of(
+            Schema.Field.of("name", Schema.FieldType.STRING),
+            Schema.Field.of("phone", Schema.FieldType.STRING));
+
+    // Create User Data Collections
+    final List<Row> emailUsers =
+        Arrays.asList(
+            Row.withSchema(emailSchema).addValue("person1").addValue("person1@example.com").build(),
+            Row.withSchema(emailSchema).addValue("person2").addValue("person2@example.com").build(),
+            Row.withSchema(emailSchema).addValue("person3").addValue("person3@example.com").build(),
+            Row.withSchema(emailSchema)
+                .addValue("person4")
+                .addValue("person4@example.com")
+                .build());
+
+    final List<Row> phoneUsers =
+        Arrays.asList(
+            Row.withSchema(phoneSchema).addValue("person1").addValue("111-222-3333").build(),
+            Row.withSchema(phoneSchema).addValue("person2").addValue("222-333-4444").build(),
+            Row.withSchema(phoneSchema).addValue("person3").addValue("444-333-4444").build(),
+            Row.withSchema(phoneSchema).addValue("person4").addValue("555-333-4444").build());
+
+    // [END SchemaJoinPatternCreate]
+
+    PCollection<String> actualFormattedResult =
+        Snippets.SchemaJoinPattern.main(p, emailUsers, phoneUsers, emailSchema, phoneSchema);
```

Comment:
As this is example snippet code, consider if it would be easier for the reader to have the code be inlined here rather than be abstracted in a class.